src/lookahead/manager.rs

```
@@ -47,9 +74,26 @@ impl LookaheadManager {
         }
     }
 
-    pub fn get_next_elected_preconfer(&self) -> Option<LookaheadEntry> {
+    fn get_next_elected_preconfer(&self) -> Option<LookaheadEntry> {
         self.lookahead.get_next_elected_preconfer()
     }
+
+    pub fn get_url(&self) -> Result<Url> {
+        match self.get_next_elected_preconfer() {
+            None => bail!("no lookahead provider found"),
+            Some(entry) => match &self.url_provider {
+                UrlProvider::LookaheadEntry => {
+                    Ok(Url::from_str(&entry.url).expect("not a valid url"))
```

i dont think we should panic here, eg. if there is a malformed string coming from the provider. Let's return an error and log the wrong string